WalletWasabi/Extensions/RPCClientExtensions.cs

```diff
@@ -59,89 +64,71 @@ private static EstimateSmartFeeResponse SimulateRegTestFeeEstimation(int confirm
+		/// </summary>
 		public static async Task<AllFeeEstimate> EstimateAllFeeAsync(this IRPCClient rpc, EstimateSmartFeeMode estimateMode = EstimateSmartFeeMode.Conservative, bool simulateIfRegTest = false)
 		{
 			var rpcStatus = await rpc.GetRpcStatusAsync(CancellationToken.None).ConfigureAwait(false);
 			var mempoolInfo = await rpc.GetMempoolInfoAsync().ConfigureAwait(false);
 			var sanityFeeRate = mempoolInfo.GetSanityFeeRate();
-			var estimations = await rpc.EstimateHalfFeesAsync(new Dictionary<int, int>(), 2, 0, Constants.SevenDaysConfirmationTarget, 0, sanityFeeRate, estimateMode, simulateIfRegTest).ConfigureAwait(false);
-			var allFeeEstimate = new AllFeeEstimate(estimateMode, estimations, rpcStatus.Synchronized);
-			return allFeeEstimate;
-		}
 
-		public static async Task<RpcStatus> GetRpcStatusAsync(this IRPCClient rpc, CancellationToken cancel)
-		{
-			try
-			{
-				var bci = await rpc.GetBlockchainInfoAsync().ConfigureAwait(false);
-				cancel.ThrowIfCancellationRequested();
-				var pi = await rpc.GetPeersInfoAsync().ConfigureAwait(false);
+			var estmimations = (simulateIfRegTest && rpc.Network == Network.RegTest)
+				? SimulateRegTestFeeEstimation(estimateMode)
+				: await GetFeeEstimationsAsync(rpc, estimateMode);
 
-				return RpcStatus.Responsive(bci.Headers, bci.Blocks, pi.Length);
-			}
-			catch (Exception ex) when (ex is not OperationCanceledException and not TimeoutException)
-			{
-				Logger.LogTrace(ex);
-				return RpcStatus.Unresponsive;
-			}
+			return new AllFeeEstimate(
+				estimateMode,
+				estmimations,
+				rpcStatus.Synchronized);
 		}
 
-		private static async Task<Dictionary<int, int>> EstimateHalfFeesAsync(this IRPCClient rpc, IDictionary<int, int> estimations, int smallTarget, int smallTargetFee, int largeTarget, int largeTargetFee, FeeRate sanityFeeRate, EstimateSmartFeeMode estimateMode = EstimateSmartFeeMode.Conservative, bool simulateIfRegTest = false)
+		private static async Task<Dictionary<int, int>> GetFeeEstimationsAsync(IRPCClient rpc, EstimateSmartFeeMode estimateMode)
 		{
-			var newEstimations = new Dictionary<int, int>();
-			foreach (var est in estimations)
-			{
-				newEstimations.TryAdd(est.Key, est.Value);
-			}
+			var batchClient = rpc.PrepareBatch();
 
-			if (Math.Abs(smallTarget - largeTarget) <= 1)
-			{
-				return newEstimations;
-			}
+			var rpcFeeEstimationTasks = Constants.ConfirmationTargets
+				.Select(target => batchClient.EstimateSmartFeeAsync(target, estimateMode))
+				.ToList();
+
+			await batchClient.SendBatchAsync().ConfigureAwait(false);
+
+			var allTask = Task.WhenAll(rpcFeeEstimationTasks);
 
-			if (smallTargetFee == 0)
+			try
 			{
-				var smallTargetFeeResult = await rpc.EstimateSmartFeeAsync(smallTarget, sanityFeeRate, estimateMode, simulateIfRegTest).ConfigureAwait(false);
-				smallTargetFee = (int)Math.Ceiling(smallTargetFeeResult.FeeRate.SatoshiPerByte);
-				newEstimations.TryAdd(smallTarget, smallTargetFee);
+				await allTask;
```

Comment:
Also possible to do:

```suggestion
				await Task.WhenAll(rpcFeeEstimationTasks).ConfigureAwait(false);
```

(and remove `allTask`)

Reply:
Yes, I don't remember why I did it in the way i did it but yes, it makes more sense as you suggest.

Reply:
Done